backend/debug/compute_ranges.ml

```diff
@@ -267,17 +302,48 @@ module Make (S : Compute_ranges_intf.S_functor) = struct
       |> handle case_2c Close_subrange_one_byte_after
     in
     let must_restart =
-      if S.must_restart_ranges_upon_any_change ()
-         && match actions with [] -> false | _ :: _ -> true
-      then KS.inter opt_available_across_prev_insn available_before
-      else KS.empty
+      KS.of_list []
+      (* CR mshinwell: it's unclear that we need this for lldb any more *)
+      (* if S.must_restart_ranges_upon_any_change () && match actions with [] ->
+         false | _ :: _ -> true then KS.inter opt_available_across_prev_insn
+         available_before else KS.of_list [] *)
     in
     actions, must_restart
 
-  let rec process_instruction t (fundecl : L.fundecl)
-      ~(first_insn : L.instruction) ~(insn : L.instruction)
+  let actions_at_instruction ~(insn : L.instruction)
+      ~(prev_insn : L.instruction option) ~known_available_after_prev_insn =
+    let available_before = S.available_before insn in
+    let available_across = S.available_across insn in
+    if debug
+    then
+      Format.eprintf "canonicalised available_before:@ %a\n"
+        (Misc.Stdlib.Option.print KS.print)
+        available_before;
+    if debug
+    then
+      Format.eprintf "canonicalised available_across:@ %a\n"
+        (Misc.Stdlib.Option.print KS.print)
+        available_across;
+    match available_before with
+    | None ->
+      (* If availability information isn't known for the current instruction,
+         just skip to the next instruction. *)
+      [], KS.of_list []
+    | Some available_before -> (
+      (* CR mshinwell: maybe we could try to compute [available_across]? In what
+         circumstances is this missing anyway? *)
+      match available_across with
```

Comment:
I think it would read better if there was a single match on both `available_before` and `available_across`.

Reply:
Fixed